### PR TITLE
updates for `_dbm._dbm`

### PR DESCRIPTION
**Pull request by tungol**
related to https://github.com/python/typeshed/issues/13016

You can get access to the type this way:

```python
import dbm.ndbm

with dbm.ndbm.open("./tmpdbm", "n") as f:
    dbm_type = type(f)
```

the existing type for `__new__` and `__init__` is odd. They do exist at runtime (at least for 3.8+), but they're just inherited from object. However the type can't be instantiated, so it was possibly an attempt to model that?

```
>>> dbm_type.__new__
<built-in method __new__ of type object at 0x10345e6b8>
>>> dbm_type.__init__
<slot wrapper '__init__' of 'object' objects>
>>> dbm_type()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: cannot create '_dbm.dbm' instances
```

I'm not sure if there's any standard way to represent that behavior in typeshed. stubtest only complained about `__new__`.

### CODE REVIEW

File: stdlib/_dbm.pyi [hunk 1]
```diff
@@ -22,20 +24,16 @@ if sys.platform != "win32":
...
-        def get(self, k: _KeyType) -> bytes | None: ...
+        def get(self, k: _KeyType, /) -> bytes | None: ...
         @overload
-        def get(self, k: _KeyType, default: _T) -> bytes | _T: ...
+        def get(self, k: _KeyType, default: _T, /) -> bytes | _T: ...
         def keys(self) -> list[bytes]: ...
-        def setdefault(self, k: _KeyType, default: _ValueType = ...) -> bytes: ...
-        # Don't exist at runtime
-        __new__: None  # type: ignore[assignment]
-        __init__: None  # type: ignore[assignment]
```
Comment (AlexWaygood) [inline stdlib/_dbm.pyi:38]: I think I'd vote for leaving these as they are, unless they're causing a concrete problem somewhere. If we do end up figuring out a way to run stubtest on this class, we can just allowlist them.

I don't know if mypy and pyright _actually_ infer that you can't create arbitrary instances of this class as a result of these assignments... but logically they _should_. I think it's the best we can do.

Comment (tungol) [reply stdlib/_dbm.pyi]: Makes sense to me. As a situation, it feels very similar to types that can't be subclassed. It's too bad we don't have a decorator like @final to handle this type of thing.

File: stdlib/_dbm.pyi [hunk 2]
```diff
@@ -22,18 +24,17 @@ if sys.platform != "win32":
...
-        def get(self, k: _KeyType) -> bytes | None: ...
+        def get(self, k: _KeyType, /) -> bytes | None: ...
         @overload
-        def get(self, k: _KeyType, default: _T) -> bytes | _T: ...
+        def get(self, k: _KeyType, default: _T, /) -> bytes | _T: ...
         def keys(self) -> list[bytes]: ...
-        def setdefault(self, k: _KeyType, default: _ValueType = ...) -> bytes: ...
-        # Don't exist at runtime
+        def setdefault(self, k: _KeyType, default: _ValueType = ..., /) -> bytes: ...
+        # This isn't true, but the class can't be instansiated. See #13024
```
Comment (AlexWaygood) [inline stdlib/_dbm.pyi:37]:
```suggestion
        # This isn't true, but the class can't be instantiated. See #13024
```

**Review by AlexWaygood — APPROVED** [review]
thanks!

**Pull request merged**